test/algos/consumers/test_sync_wait.cpp

```diff
@@ -154,14 +174,24 @@ optional<tuple<std::string>> tag_invoke(
   return {res};
 }
 
-optional<tuple<std::string>> tag_invoke(decltype(sync_wait), just_string_sender_t s) {
-  std::string res;
-  auto op = ex::connect(std::move(s), expect_value_receiver_ex{&res});
-  ex::start(op);
-  CHECK(res == "hello");
-  // change the string
-  res = "ciao";
-  return {res};
+struct my_other_string_sender_t {
+  std::string str_;
+
+  using completion_signatures = typename decltype(ex::just(std::string{}))::completion_signatures;
```

Comment:
I would prefer if tests did not assume implementation details like this. Either be explicit about the completion signatures, or use `completion_signatures_of_t`.

Reply:
```suggestion
  using completion_signatures = ex::completion_signatures_of_t<decltype(ex::just(std::string{}))>;
```